pydicom/dataset.py

```diff
@@ -1564,6 +1568,159 @@ def _do_pixel_data_conversion(
 
         self._pixel_id = get_image_pixel_ids(self)
 
+    def compress(
+        self,
+        uid: str,
```

Comment:
`transfer_syntax_uid`? Or would that be too long for convenience?

Reply:
Yeah it's probably better. It's required anyway so people are unlikely to be typing it out all the time.